specs/PrintToPdf.md

```diff
@@ -0,0 +1,294 @@
+  [propget] HRESULT ScaleFactor([out, retval] double* scaleFactor);
+
+  /// Sets the `ScaleFactor` property.
+  [propput] HRESULT ScaleFactor([in] double scaleFactor);
+
+  /// The page width in inches. The default width is 8.5 inches.
+  [propget] HRESULT PageWidth([out, retval] double* pageWidth);
+
+  /// Sets the `PageWidth` property.
+  [propput] HRESULT PageWidth([in] double pageWidth);
+
+  /// The page height in inches. The default height is 11 inches.
+  [propget] HRESULT PageHeight([out, retval] double* pageHeight);
+
+  /// Sets the `PageHeight` property.
+  [propput] HRESULT PageHeight([in] double pageHeight);
+
+  /// The top margin in inches. The default is 1 cm, or ~0.4 inches.
+  [propget] HRESULT MarginTop([out, retval] double* marginTop);
+
+  /// Sets the `MarginTop` property.
+  [propput] HRESULT MarginTop([in] double marginTop);
+
+  /// The bottom margin in inches. The default is 1 cm, or ~0.4 inches.
+  [propget] HRESULT MarginBottom([out, retval] double* marginBottom);
+
+  /// Sets the `MarginBottom` property.
+  [propput] HRESULT MarginBottom([in] double marginBottom);
+
+  /// The left margin in inches. The default is 1 cm, or ~0.4 inches.
+  [propget] HRESULT MarginLeft([out, retval] double* marginLeft);
+
+  /// Sets the `MarginLeft` property.
+  [propput] HRESULT MarginLeft([in] double marginLeft);
+
+  /// The right margin in inches. The default is 1 cm, or ~0.4 inches.
+  [propget] HRESULT MarginRight([out, retval] double* marginRight);
+
+  /// Set the `MarginRight` property.
+  [propput] HRESULT MarginRight([in] double marginRight);
+
+  /// `TRUE` if backgrounds should be printed. The default value is `FALSE`.
+  [propget] HRESULT ShouldPrintBackgrounds(
+      [out, retval] BOOL* shouldPrintBackgrounds);
+
+  /// Set the `ShouldPrintBackgrounds` property.
+  [propput] HRESULT ShouldPrintBackgrounds([in] BOOL shouldPrintBackgrounds);
+
+  /// `TRUE` if only the selection should be printed. The default value is `FALSE`.
+  [propget] HRESULT ShouldPrintSelectionOnly(
+      [out, retval] BOOL* shouldPrintSelectionOnly);
+
+  /// Set the `ShouldPrintSelectionOnly` property.
+  [propput] HRESULT ShouldPrintSelectionOnly(
+      [in] BOOL shouldPrintSelectionOnly);
+
+  /// `TRUE` if header and footer should be printed. The default value is `FALSE`.
+  [propget] HRESULT ShouldPrintHeaderFooter(
+      [out, retval] BOOL* shouldPrintHeaderFooter);
+
+  /// Set the `ShouldPrintHeaderFooter` property.
+  [propput] HRESULT ShouldPrintHeaderFooter([in] BOOL shouldPrintHeaderFooter);
+
+  /// The title in the header if `ShouldPrintHeaderFooter` is `TRUE`.
+  [propget] HRESULT HeaderTitle([out, retval] LPWSTR* headerTitle);
+
+  /// Set the `HeaderTitle` property.
+  [propput] HRESULT HeaderTitle([in] LPCWSTR headerTitle);
+
+  /// The URI in the footer if `ShouldPrintHeaderFooter` is `TRUE`.
+  [propget] HRESULT FooterUri([out, retval] LPWSTR* footerUri);
```

Comment:
Similar question about the header but for the footer if the uri is empty string or null